iothub/device/src/Authentication/HsmAuthentication/Transport/HttpRequestResponseSerializer.cs

```diff
@@ -120,9 +120,17 @@ internal static async Task<HttpResponseMessage> DeserializeResponseAsync(HttpBuf
                     {
                         if (!long.TryParse(headerValue, out long contentLength))
                         {
-                            throw new HttpRequestException($"Header value is invalid for {headerName}.");
+                            throw new IotHubClientException($"Header value is invalid for {headerName}.", IotHubClientErrorCode.NetworkErrors);
+                        }
+
+                        try
+                        {
+                            await httpResponse.Content.LoadIntoBufferAsync(contentLength).ConfigureAwait(false);
+                        }
+                        catch (HttpRequestException ex)
+                        {
+                            throw new IotHubClientException(ex.Message, IotHubClientErrorCode.NetworkErrors, ex);
```

Comment:
is transient?

Reply:
Implied from network errors?

Reply:
Yes, `IotHubClientErrorCode.NetworkErrors` will be taken as transient within ctor of `IotHubClientException`